bot/common.go

```diff
@@ -21,7 +21,11 @@ func generateUpcomingContestsMessage(clistService *clist.Service, startFrom, sta
 	buffer.WriteString("\n")
 	var res []string
 	for _, contest := range contests {
-		str := fmt.Sprintf("\n- %s. Starts at %s. Link: %s", contest.Name, contest.StartDate.In(tz).Format("Jan 2 15:04 MST"), contest.Link)
+		str := ""
+		if buffer.Len() > 0 {
```

Comment:
`buffer.Len()` will never be 0, so this will always be executed.

Better approach is to have it like this:

```go
str := fmt.Sprintf("- ...") // without \n
if buffer.Len() + len(str) + 1 > limit {
    res = append(res, buffer.String())
    buffer = *bytes.NewBufferString(str)
} else {
    buffer.WriteString("\n")
    buffer.WriteString(str)
}
```